### DISCUSSION TITLE
Forbidden Access to threadOf Attribute

### DISCUSSION THREAD

**Discussion by StoCreator** [VirtusLab-Open-Source/strapi-plugin-comments#200] [Q&A] [answered]
Hello everybody,

we have a problem with the plugin. Whenever we try to query the comments plugin we get a forbidden access message. This only happens when we try to query the field "threadOf".
We already had a look at the admin panel concerning the authorization permission, but nothing helps.

Is there anybody who knows how to deactivate authoritazion for the comments plugin or has other ideas?
Thanks in advance.

**Comment by cyp3rius** [comment]
Can you please provide a query examples and responses?

That would be good if you'll check as well if proper endpoints got applied access rights for public / authorized.

**Reply by StoCreator** [reply]
Hello cyp3rius,

thank you for your answer. 
First of all we are using Strapi 4 and graphql as query language.

Here is the schema of the usercomments that comes out of our docs: 
![Unbenannt](https://user-images.githubusercontent.com/124581751/230318911-48c6823c-f9f5-4465-bc4b-120015d80d8e.PNG)

This is an example how we query the object:
![1](https://user-images.githubusercontent.com/124581751/230320387-652f796a-f515-4a90-95bf-a3df508d10d8.PNG)

That is what we gat as a result:
![Unbenann1t](https://user-images.githubusercontent.com/124581751/230320286-63eeee4d-d445-4121-8b72-26b8670b3656.PNG)

In the custom resolver for usercomments we return the object like the following:
![Unbenandnt](https://user-images.githubusercontent.com/124581751/230322255-430c2678-d7ec-4266-96d0-6f7b33adf696.PNG)
We already set the user authorization for usercomments to false in the resolverconfig.

The Forbidden Error message only appears when we try to query the threadOf object. We can query everything else without getting the error message.

Thank you in advance for  your help. Please let us know if you need more information.

**Reply by cyp3rius** [reply]
It's hard to say as you already mentioned that you've got custom resolver so the error is caused by something inside. What you can do is:
- log your data structures before `return` and check if you've got all properties returned (including `threadOf`)
- try to use the build in query and just provide the `authorUser` prop, that should do the same as your custom one

**Comment by StoCreator** [comment, marked as answer]
Thank you for your answer and sorry for the late reply. 

We now solved the problem by adding the following Code to the resolversConfig:

![Unbenannt](https://user-images.githubusercontent.com/124581751/233290439-e924d03a-cd14-44a2-9588-3529d0eb271e.PNG)
